src/mpid/ch4/src/ch4_init.c

```diff
@@ -526,29 +526,11 @@ int MPID_Init_local(int requested, int *provided)
     mpi_errno = MPIR_pmi_init();
     MPIR_ERR_CHECK(mpi_errno);
 
-    int err;
-    MPID_Thread_mutex_create(&MPIDIU_THREAD_PROGRESS_MUTEX, &err);
-    MPIR_Assert(err == 0);
-
-    MPID_Thread_mutex_create(&MPIDIU_THREAD_UTIL_MUTEX, &err);
-    MPIR_Assert(err == 0);
-
-    MPID_Thread_mutex_create(&MPIDIU_THREAD_MPIDIG_GLOBAL_MUTEX, &err);
-    MPIR_Assert(err == 0);
-
-    MPID_Thread_mutex_create(&MPIDIU_THREAD_SCHED_LIST_MUTEX, &err);
-    MPIR_Assert(err == 0);
-
-    MPID_Thread_mutex_create(&MPIDIU_THREAD_TSP_QUEUE_MUTEX, &err);
-    MPIR_Assert(err == 0);
-
-#ifdef HAVE_LIBHCOLL
-    MPID_Thread_mutex_create(&MPIDIU_THREAD_HCOLL_MUTEX, &err);
-    MPIR_Assert(err == 0);
-#endif
-
-    MPID_Thread_mutex_create(&MPIDIU_THREAD_DYNPROC_MUTEX, &err);
-    MPIR_Assert(err == 0);
+    for (int i = 0; i < MAX_CH4_MUTEXES; i++) {
+        int err;
+        MPID_Thread_mutex_create(&MPIDI_global.m[i], &err);
+        MPIR_Assert(err == 0);
+    }
```

Comment:
Compared with the original code, `MPIDIU_THREAD_HCOLL_MUTEX` is now always initialized no matter `HAVE_LIBHCOLL` is defined or not. But I think it is fine as it is just a mutex which is initialized but never used. To reduce potential concern from other reader, maybe we could have a brief comment here to explain such cases.

Reply:
Comment added.